src/app/navigation/navigation-item-config.ts

```diff
@@ -0,0 +1,70 @@
+/**
+ * A config containing properties for navigation items
+ */
+export class NavigationItemConfig {
+
+  /**
+   * Title for the navigation item
+   */
+  title: string;
+
+  /**
+   * The icon class to use for icons displayed to the left of text
+   */
+  iconClass?: string;
+
+  /**
+   * Link to navigate to
+   */
+  href?: string;
+
+  /**
+   * Badges to display information about the navigation item
+   */
+  badges?: any[];
+
+  /**
+   * Navigation children (used for secondary and tertiary navigation)
+   */
+  children?: NavigationItemConfig[];
+
+  /**
+   * Indicate if the item should be active on load
+   */
+  active?: boolean;
+
+  /**
+   * Track the active state of the navigation item
+   */
+  isActive?: boolean;
```

Comment:
Don't care for the isActive and isHover names. Although I've seen 'is' prefixed for some attribute names, we've reserved this convention for methods (i.e., something that tests state where a boolean is returned).

I'm thinking trackActiveState and trackHoverState may be more appropriate here?